connection/kubernetes.go

```diff
@@ -0,0 +1,42 @@
+package connection
+
+import (
+	"fmt"
+
+	"github.com/flanksource/duty/models"
+	"github.com/flanksource/duty/types"
+)
+
+// +kubebuilder:object:generate=true
+type KubernetesConnection struct {
+	ConnectionName string       `json:"connection,omitempty"`
+	KubeConfig     types.EnvVar `json:"kubeconfig,omitempty"`
+}
+
+func (t KubernetesConnection) ToModel() models.Connection {
+	return models.Connection{
+		Type:        models.ConnectionTypeKubernetes,
+		Certificate: t.KubeConfig.ValueStatic,
+	}
+}
+
+func (t *KubernetesConnection) Populate(ctx ConnectionContext) error {
+	if t.ConnectionName != "" {
+		connection, err := ctx.HydrateConnectionByURL(t.ConnectionName)
+		if err != nil {
+			return err
+		} else if connection == nil {
+			return fmt.Errorf("connection[%s] not found", t.ConnectionName)
+		}
+
+		t.KubeConfig.ValueStatic = connection.Certificate
+	}
+
+	if v, err := ctx.GetEnvValueFromCache(t.KubeConfig, ctx.GetNamespace()); err != nil {
```

Comment:
The namespace of the playbook will unlikely match the namespace of the scrapeconfig

Reply:
namespace is set in the context in here

https://github.com/flanksource/duty/pull/1181/files#diff-e3e90426a3460155dcf682688699c1c9b2b2cb46ed156387b6efc1675808cfefR107